bootloader/src/pyi_win32_utils.c

```diff
@@ -491,5 +491,34 @@ int pyi_win32_is_symlink(const wchar_t *path)
     return 0;
 }
 
+/* Check if the given path is just a drive letter */
+int pyi_win32_is_drive_root(const wchar_t *path)
+{
+    /* For now, handle just drive letter, optionally followed by the path separator.
+       E.g., "C:" or "Z:\".
+     */
+    size_t len;
+
+    len = wcslen(path);
+    if (len == 2 || len == 3) {
+        /* First character must be a letter */
+        if (!iswalpha(path[0])) {
+            return 0;
+        }
+        /* Second character must be the colon */
+        if (path[1] != L':') {
+            return 0;
+        }
+        /* Third character, if present, must be the Windows directory separator */
+        if (len > 2 && (path[2] != L'\\')) {
```

Comment:
Could `path[2]` not also be `'/'`?

Reply:
It *shouldn't* be, because our Windows-specific bootloader codepaths assume Windows separators (e.g., [here](https://github.com/pyinstaller/pyinstaller/blob/1cadf324421d32759b205d14087391f90d6ef8ae/bootloader/src/pyi_utils.c#L277)). But with the world we live in, it *could* well be, although it takes some effort to set up such scenario - for example, if the path (`c:/`) is stored to environment variable, and the environment variable is passed as `runtime_tmpdir` via spec file. 

If it were passed directly on command-line, it would end up expanded, and then TOC would normalize it:
https://github.com/pyinstaller/pyinstaller/blob/1cadf324421d32759b205d14087391f90d6ef8ae/PyInstaller/archive/writers.py#L262-L268

(As a side note, the `nm` passed to the `normpath` is actually not just the path, but whole run-time option, e.g., `'pyi-runtime-tmpdir c:/'`. So `normpath` seems to end up stripping the trailing separator, even though it otherwise would not.)

But, it looks like that even if you manage to smuggle a `c:/` in and we don't treat it as a drive root because of separator mismatch, that `_wfullpath` call changes it from `c:/` to `c:\` and all is well. So I'd prefer to keep it this way rather than treat `c:/` as drive root and then have to deal with unexpected separator somewhere else...